### DISCUSSION TITLE
how can I dispatch an event from controller to livewire component

### DISCUSSION THREAD

**Discussion by matiazar** [livewire/livewire#9011] [1. Help]
Hi

I have a livewire component that I want to update (render again) every time a new transaction comes in db..

I used the     #[On('event-name')] attribute in component.. but I dont understand how to send this event from a controller.

thanks in advance

**Comment by danie-ramdhani** [comment]
You can't dispatch from laravel controller. Try dispatch it from the view

```
<button @click="$dispatch('post-created')">...</button>
```

https://livewire.laravel.com/docs/events#dispatching-livewire-events-from-alpine

**Reply by matiazar** [reply]
but I need to dispatch it after some events occurs... not with a button

**Comment by matiazar** [comment]
I did it... in other way
I dont like to use a button... because is an automatic event not a manual one

**Reply by mikailhamzat** [reply]
which way. care to share?

**Reply by matiazar** [reply]
I created a free Pusher account

and then in the controller I have sent a new broadcast event

event(new BroadcastEvent($data));

BroadcastEvent is an event class that send a broadcast to pusher

and you need to listen to this channel that you have configured in the event in the front with JS

var channel = window.Echo.channel('channel-name');
channel.listen('.event-name', function (data) {
    Livewire.dispatch('event');
});


and in the livewire component before the function that you want to re-render

#[On('event')]

and this will refresh the component to the user and to all other users listining to the same channel/event

**Reply by lrljoe** [reply]
Depends on your use case, but that may bring more problems than it solves longer term.

For example - your app is popular, and it ends up with 1 transaction every 3 seconds.  That's going to become unusable for people if it's constantly refreshing, and it'll introduce heavy load server-side.

Consider how you throttle that behaviour, and how you target it.  And what data you're throwing via pusher.

**Reply by mikailhamzat** [reply]
> I created a free Pusher account
> 
> and then in the controller I have sent a new broadcast event
> 
> event(new BroadcastEvent($data));
> 
> BroadcastEvent is an event class that send a broadcast to pusher
> 
> and you need to listen to this channel that you have configured in the event in the front with JS
> 
> var channel = window.Echo.channel('channel-name'); channel.listen('.event-name', function (data) { Livewire.dispatch('event'); });
> 
> and in the livewire component before the function that you want to re-render
> 
> #[On('event')]
> 
> and this will refresh the component to the user and to all other users listining to the same channel/event

Thank you for this. I also got his solution from Claude but i thought it was an overkill. Its actually very easy to set up. Thanks for confirming that it's working.

**Reply by mikailhamzat** [reply]
> Depends on your use case, but that may bring more problems than it solves longer term.
> 
> For example - your app is popular, and it ends up with 1 transaction every 3 seconds. That's going to become unusable for people if it's constantly refreshing, and it'll introduce heavy load server-side.
> 
> Consider how you throttle that behaviour, and how you target it. And what data you're throwing via pusher.

Perhaps you can use private channel to only dispatch event for the affected user only
